src/soc/intel/alderlake/romstage/fsp_params.c

```diff
@@ -180,6 +180,18 @@ static void fill_fspm_cpu_params(FSP_M_CONFIG *m_cfg,
 	m_cfg->PrmrrSize = get_valid_prmrr_size();
 	m_cfg->EnableC6Dram = config->enable_c6dram;
 	m_cfg->HyperThreading = get_uint_option("hyper_threading", CONFIG(FSP_HYPERTHREADING));
+
+	if (CONFIG(DRIVERS_EFI_VARIABLE_STORE) && CONFIG(FSP_HYPERTHREADING))
+		m_cfg->HyperThreading = get_hyper_threading_option();
+
+	m_cfg->ActiveCoreCount = get_active_core_count_option();
+	m_cfg->ActiveSmallCoreCount = get_active_small_core_count_option();
+
+	if (m_cfg->ActiveCoreCount == 0 && m_cfg->ActiveSmallCoreCount == 0) {
```

Comment:
You've mentioned in Jira that disabling all P-cores doesn't work, while this check suggests that it's possible to disable all of them.  So which is it?

Reply:
It doesn't but there are SKUs that only have E-cores. And this check is purely added to avoid a soft-brick by injecting invalid values into EFIVAR store. Disabling all cores should never be possible, but variables are just an external input, which may be malicious/invalid.

Reply:
I think we won't get around it without actually adding the SKU table to coreboot and checking the settings against the number of cores of given type...

Reply:
Added the SKU table and refactored the checks